components/lib/crash/src/test/java/mozilla/components/lib/crash/notification/CrashNotificationTest.kt

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.crash.notification
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.CrashReporter
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Shadows.shadowOf
+
+@RunWith(AndroidJUnit4::class)
+class CrashNotificationTest {
+    @Test
+    fun shouldShowNotificationInsteadOfPrompt() {
+        val nonFatalNativeCrash = Crash.NativeCodeCrash(
+            minidumpPath = "",
+            minidumpSuccess = true,
+            extrasPath = "",
+            isFatal = false
+        )
+
+        assertFalse(CrashNotification.shouldShowNotificationInsteadOfPrompt(nonFatalNativeCrash, sdkLevel = 21))
```

Comment:
nit: What do you think about generalizing this assertion (with using failing message)? Something like:

```kotlin
verifyNotificationInsteadOfPrompt(
    crash = crash,
	shouldPreferIf = { sdkVersion -> sdkVersion >= 29 }
)

fun verifyNotificationInsteadOfPrompt(...) {
    for (sdkVersion in minSdk..targetSdk) {
	    val expected = shouldPreferIf(sdkVersion)
	    val actual = shouldShowNotificationInsteadOfPrompt(nonFatalNativeCrash, sdkLevel = sdkVersion)
		assertEquals("Verifying for SDK: $sdkVersion", expected, actual)
	}
}
```

This way has it's own downsides but it would be nice to test full range of app enabled sdk versions.

Reply:
I think some repetition is okay in tests. I don't want to end up in a situation where my test code contains complex logic so that it requires tests itself :)